src/controls/Controls.spec.js

```diff
@@ -1 +1,87 @@
-// Test away!
+import React from 'react';
+import { render, fireEvent, cleanup } from 'react-testing-library';
+import renderer from 'react-test-renderer';
+
+import Controls from './Controls';
+
+beforeEach(cleanup);
+
+describe('<Controls />', () => {
+  it('matches snapshot', () => {
+    const tree = renderer.create(<Controls />).toJSON();
+    expect(tree).toMatchSnapshot();
+  });
+
+  describe('Lock/Unlock Button', () => {
+    it('diplays lock gate if gate is unlocked', () => {
+      const locked = false;
+      const { getByText } = render(<Controls locked={locked} />);
+      getByText(/^lock gate$/i);
+    });
+
+    it('diplays unlock gate if gate is locked', () => {
+      const locked = true;
+      const { getByText } = render(<Controls locked={locked} />);
+      getByText(/^unlock gate$/i);
+    });
+
+    it('is enabled if gate is closed', () => {
+      const closed = true;
+      const mockToggleLocked = jest.fn();
+      const { getByText } = render(
+        <Controls closed={closed} toggleLocked={mockToggleLocked} />
+      );
+      fireEvent.click(getByText(/^lock gate$/i));
+      expect(mockToggleLocked).toHaveBeenCalledTimes(1);
+    });
+
+    it('is disabled if gate is opened', () => {
+      const closed = false;
+      const mockToggleLocked = jest.fn();
+      const { getByText } = render(
+        <Controls closed={closed} toggleLocked={mockToggleLocked} />
+      );
+      fireEvent.click(getByText(/^lock gate$/i));
+      expect(mockToggleLocked).toHaveBeenCalledTimes(0);
```

Comment:
yes, this was the use case for mocks that I was visualizing for this app as well. the non-mock way to check this same things is to verify the disabled status of the button itself in either state, either by running an `expect(button.disabled)` to be either truthy or falsy, since disabled is a property on every button that defaults as false, or to use the jest-dom/extend-expect library and use the `.toBeDisabled()/.toBeEnabled()` matcher methods